src/Http/Http.Extensions/test/ParameterBindingMethodCacheTests.cs

```diff
@@ -1069,6 +1125,30 @@ public static void BindAsync(HttpContext context)
             => throw new NotImplementedException();
     }
 
+    private class BindAsyncFromImplicitStaticAbstractInterface : IBindableFromHttpContext<BindAsyncFromImplicitStaticAbstractInterface>
+    {
+        public static ValueTask<BindAsyncFromImplicitStaticAbstractInterface?> BindAsync(HttpContext context, ParameterInfo parameter)
+        {
+            return ValueTask.FromResult<BindAsyncFromImplicitStaticAbstractInterface?>(new());
+        }
+    }
+
+    private class BindAsyncFromExplicitStaticAbstractInterface : IBindableFromHttpContext<BindAsyncFromExplicitStaticAbstractInterface>
+    {
+        static ValueTask<BindAsyncFromExplicitStaticAbstractInterface?> IBindableFromHttpContext<BindAsyncFromExplicitStaticAbstractInterface>.BindAsync(HttpContext context, ParameterInfo parameter)
```

Comment:
Can we add test cases returning nullable and non-nullable structs? What if you implement both `IBindableFromHttpContext<MyStruct>` and `IBindableFromHttpContext<Nullable<MyStruct>>`?

Reply:
I'm not sure how to even do this as the compiler won't let me create a struct the implements `IBindableFromHttpContext<MyStruct>` with a `ValueTask<MyStruct?> BindAsync` method:

<img width="1143" alt="image" src="https://user-images.githubusercontent.com/249088/163039698-34b22d5c-8f14-49ce-9645-ee28ab91e89d.png">

<img width="1142" alt="image" src="https://user-images.githubusercontent.com/249088/163039765-1a27cf7e-6e34-437c-a326-223a0a9ff08c.png">

Reply:
I think we're going to have to remove the `where TSelf : IBindableFromHttpContext<TSelf>` generic constraint. The constraint was already not perfect because `TSelf` could be any `IBindableFromHttpContext<TSelf>` and not just the type implementing it.

Reply:
We spoke about this more and decided we'll leave the constraint in place for now as the scenario of wanting to bind a nullable struct seems super edge-case and if ppl want to do that, they can simply not use the interface, or define their own interface without the constraint, as we'll still find and use the `BindAsync` method that way anyway. Plus we can decide the remove the constraint later if we really want to and it won't be a breaking change.

Reply:
I'm fine with leaving the `where TSelf : IBindableFromHttpContext<TSelf>` constraint, but then I think we should add a `where T : class?` constraint since we rely on null returns to support optionality with `BindAsync`. There's still always the out of implementing the method publicly without the interface.